subprojects/gradle/room-config/src/main/kotlin/com/avito/android/plugin/incremental/RoomIncrementalKaptSupportChecker.kt

```diff
@@ -0,0 +1,53 @@
+package com.avito.android.plugin.incremental
+
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.internal.KaptGenerateStubsTask
+import java.net.URLClassLoader
+import javax.annotation.processing.AbstractProcessor
+
+internal class RoomIncrementalKaptSupportChecker(
+    private val project: Project
+) {
+
+    fun isSupported(): Boolean {
+        val processor = createRoomProcessor()
+        try {
+            return checkOfficialWay(processor)
+        } catch (e: Exception) {
+            System.err.println("Fail to check Room for KAPT incremental support in official way: %e")
+        }
+
+        try {
+            return checkReflectionWay(processor)
+        } catch (e: Exception) {
+            System.err.println("Fail to check Room for KAPT incremental support in reflection way: %e")
```

Comment:
I believe that we should fail if none is working.
Fail fast shows the problem. In this case, a user will disable the broken check.

Reply:
Not actual. Move this check from this PR to another task.